src/Middlewares/PermissionMiddleware.php

```diff
@@ -12,6 +13,20 @@ public function handle($request, Closure $next, $permission, $guard = null)
     {
         $authGuard = Auth::guard($guard);
 
+        $permissions = is_array($permission)
+            ? $permission
+            : explode('|', $permission);
+
+        // For machine-to-machine Passport clients
+        $bearerToken = $request->bearerToken();
+        if ($bearerToken) {
+            $client = ClientService::getClient($bearerToken);
```

Comment:
What about `client` build in method? [passport/Guards/TokenGuard.php#L149](https://github.com/laravel/passport/blob/b566e159c9661c49ca07b8e12a91ef86ee1bb94c/src/Guards/TokenGuard.php#L149)

I had something like this in mind:
```php
public function handle($request, Closure $next, $permission, $guard = null)
{
    $authGuard = Auth::guard($guard);

    $bearerToken = $request->bearerToken();
    if ($bearerToken && method_exists($authGuard, 'client')) {
        $user = $authGuard->client();
    }

    $user = $user ?? $authGuard->user();

    if (! $user) {
        throw UnauthorizedException::notLoggedIn();
    }

    if (! method_exists($user, 'hasAnyPermission')) {
        throw UnauthorizedException::missingTraitHasRoles($user);
    }

    $permissions = is_array($permission)
        ? $permission
        : explode('|', $permission);

    if (! $user->canAny($permissions)) {
        throw UnauthorizedException::forPermissions($permissions);
    }

    return $next($request);
}
```